invenio_oauthclient/contrib/keycloak/helpers.py

```
@@ -64,51 +102,13 @@ def get_user_info(remote, resp_token, fallback_to_endpoint=True, options=dict())
-        pubkey = _format_public_key(get_public_key(remote, realm_url))
-        alg = jwt.get_unverified_header(token)["alg"]
-
-        if not isinstance(options, dict):
-            options = {}
-
-        # consult the config whether to check the target audience
-        should_verify_aud = current_app.config.get(f"{config_prefix}_VERIFY_AUD", False)
-        expected_aud = current_app.config.get(f"{config_prefix}_AUD", None)
-
-        if should_verify_aud and (expected_aud is not None):
-            options.update({"verify_aud": True})
-        else:
-            options.update({"verify_aud": False})
-
-        # consult the config whether to check signature expiration
-        should_verify_expiration = current_app.config.get(
-            f"{config_prefix}_VERIFY_EXP", False
-        )
-
-        if should_verify_expiration:
-            options.update({"verify_exp": True})
-        else:
-            options.update({"verify_exp": False})
-
-        user_info = jwt.decode(
-            token, key=pubkey, algorithms=[alg], audience=expected_aud, options=options
-        )
-
-    except Exception as e:
-        if not fallback_to_endpoint:
-            raise OAuthKeycloakUserInfoError(
-                "Error while fetching user information: {}".format(e),
-                remote,
-                resp_token,
-            )
-
-        # as a fallback, we can still contact Keycloak's userinfo endpoint
-        # `remote.get(...)` automatically includes OAuth2 tokens in the header
-        # and the response's `data` field is a dict
-        url = current_app.config[f"{config_prefix}_USER_INFO_URL"]
-        user_info = remote.get(url).data
-
-    return user_info
+    use_token = current_app.config[f"{config_prefix}_USER_INFO_FROM_TOKEN"]
+    if use_token:
+        try:
+            return _get_user_info_from_token(remote, resp_token["id_token"], config_prefix)
+        except Exception as e:
+            pass
```

minor: should we maybe log the exception instead of just swallowing it?